newIDE/app/src/MainFrame/index.js

```diff
@@ -854,7 +854,15 @@ const MainFrame = (props: Props) => {
 
   const loadFromProject = React.useCallback(
     async (project: gdProject, fileMetadata: ?FileMetadata): Promise<State> => {
-      if (fileMetadata) {
+      let updatedFileMetadata: ?FileMetadata = fileMetadata
```

Comment:
Would need a comment explaining why the heck we would need to update the fileMetadata despite being passed to us (us = the function). Is this our job? Is this the caller job? 

I don't mind fixing something on the fly, but we need to document this in a function called "patchFileMetadataWithProjectInfoIfNeeded" to make extra clear we're doing something here and have the ability to comment why (without making the function bigger).